Yoast/javascript
# 426 create link statistics research

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds a link statistics research for the new tree parser. 

## Relevant technical choices:

* This research does not work yet with the `TreeResearcher`, since the research needs a way to access the URL of the current document.
  * I added a second argument containing the metadata to the `doResearch` method on the `Research` class. This class is a `Node`, since we planned to add it as a node to the tree.
 * However, this **will** change, since we now plan to add the data that we need for analysis as attributes to the `ParsedPaper`.

## Test instructions

This PR can be tested by following these steps:

* Run the unit tests.
* See that they pass.
* Check that the coverage of the `LinkStatistics` research is 100%.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes #426
